services/device-registry-file/src/main/java/org/eclipse/hono/deviceregistry/file/FileBasedRegistrationService.java

```diff
@@ -520,6 +527,108 @@ private JsonObject convertDevice(final String deviceId, final Device payload) {
         return identities.computeIfAbsent(tenantId, id -> new ConcurrentHashMap<>());
     }
 
+    /**
+     * Return all devices for a tenant.
+     *
+     * This implementation is for demo purpose, it does not support filtering and sorting.
+     * Moreover, the order of devices cannot be guaranteed by the underlying in-memory storage
+     * so the paging may be inconsistent.
+     */
+    @Override
+    public Future<OperationResult<List<DeviceWithId>>> searchDevices(final String tenantId, final Optional<Integer> limit, final Optional<Integer> offset,
+                                                                     final Optional<List<SortingOption>> sorting, final Optional<List<Filter>> filters, final Span span) {
+        final int actualLimit = limit.map(l -> {
+            if (l > config.getMaxPageLimit()) {
+              return config.getMaxPageLimit();
+            } else {
+              return l;
+            }
+        }).orElse(config.getDefaultPageLimit());
+        final int actualOffset = offset.orElse(0);
+
+        return Future.succeededFuture(doSearchDevices(tenantId, actualLimit, actualOffset, sorting.orElse(Collections.EMPTY_LIST), filters.orElse(Collections.EMPTY_LIST)));
+    }
+
+    private OperationResult<List<DeviceWithId>> doSearchDevices(final String tenantId, final int limit, final int offset,
```

Comment:
If I am not mistaken then you are returning `DeviceWithId` instances to the client which include the device identifier. However, the OpenAPI spec defines the returned result as an array of Device objects which do not contain an ID.

Reply:
@sophokles73 this was taken care of by @kaniyan in #2113.